spyder/api/utils.py

```diff
@@ -0,0 +1,1096 @@
+# -*- coding: utf-8 -*-
```

Comment:
I don't think this should be exposed as part of our API because it's composed purely of helper functions.

Please move the contents of this file to `utils`, finding a right place for each function in our current files. If no place can be found, please create a new `utils/folding.py` file.